pkg/network/setup/configstate.go

```diff
@@ -45,40 +45,55 @@ func NewConfigState(cc cacheCreator, vmiUID string) ConfigState {
 //
 // The discovery step can be executed repeatedly with no limitation.
 // The configuration step is allowed to run only once. Any attempt to run it again will cause a critical error.
-func (c ConfigState) Run(podIfaceName string, discoverFunc func() error, configFunc func() error) error {
-	state, err := c.read(podIfaceName)
-	if err != nil {
-		return err
-	}
+func (c ConfigState) Run(nics []podNIC, discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
+	for _, nic := range nics {
+		ifaceName := nic.podInterfaceName
+		state, err := c.read(ifaceName)
+		if err != nil {
+			return err
+		}
 
-	switch state {
-	case cache.PodIfaceNetworkPreparationStarted:
-		return neterrors.CreateCriticalNetworkError(
-			fmt.Errorf("pod interface %s network preparation cannot be restarted", podIfaceName),
-		)
-	case cache.PodIfaceNetworkPreparationFinished:
-		return nil
+		switch state {
+		case cache.PodIfaceNetworkPreparationStarted:
+			return neterrors.CreateCriticalNetworkError(
+				fmt.Errorf("pod interface %s network preparation cannot be restarted", ifaceName),
+			)
+		case cache.PodIfaceNetworkPreparationFinished:
```

Comment:
Maybe I'm missing something, but, from my perspective, the whole point for introducing state to know when CNI has finished configuring the pod interface was to prevent errors on the discover phase - i.e. kubevirt attempting to discover the pod interface before it was ready in the pod. 

Furthermore - and again I might be missing something - why do you say `netA` will not be tried again ? AFAIU, if `netB` succeeds, it'll manage to plug a new interface into the domain, which will mutate the vmi status w/ information from the domain - but the plug operation for `netA` has failed, which will cause the reconciliation loop to still spot a difference between current interface state and desired interface state, which will trigger the loop from the beginning. 

Maybe a short sync makes sense to iron out the differences here - I feel there's something I might be missing.

Reply:
Done